src/PolykeyAgent.ts

```diff
@@ -754,16 +755,19 @@ class PolykeyAgent {
     this.logger.info(`Destroying ${this.constructor.name}`);
     // DB needs to be running for dependent domains to properly clear state.
     await this.db.start();
+    // TaskManager needs to be running for Discovery to destroy
+    await this.taskManager.start({ lazy: true });
```

Comment:
This would only happen if we stop then destroy the domains... Right?

Usually when stopping we stop the db. Then if we want destroy, it still needs access to the db so it is started again for the destruction.

So the issue here is that upon destroying discovery, we have to destroy state including state the tasks system.

This doesn't affect nodes cause all tasks are already removed when the node manager is stopped?

Reply:
Yeah. All the tasks in `NodeManager` are ephemeral, they are cancelled during stopping. Here the discovery tasks persist. So we only remove them when cleaning up during destruction or starting fresh.

Reply:
Updated the comment.